handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -412,22 +412,30 @@ public String applicationProtocol() {
      *
      * @return the {@link Future} for the iniital TLS handshake if {@link #renegotiate()} was not invoked.
      *         The {@link Future} for the most recent {@linkplain #renegotiate() TLS renegotiation} otherwise.
+     * @deprecated intercept {@link SslHandshakeCompletionEvent}.
      */
+    @Deprecated
     public Future<Channel> handshakeFuture() {
```

Comment:
A user could use the events if he or she writes the sending part of the client in ChannelHandler but it's often more convenient to write from outside of ChannelHandler. e.g. a user writes a synchronous client which is part of a command line utility:

```java
    Channel ch = ...;
    sslHandler.handshakeFuture().sync();
    ch.write(firstRequest).sync();
    mainHandler.receivedMessages().take()...
```